### PR TITLE
[1034] Compact Header

### PR DESCRIPTION
**Pull request by rlaco-qualityunit**
**Changes proposed in this Pull Request**
Changes according comments from designers in figma.

**Testing instructions**
Check changes based on comments from designers in figma.

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] My commits follow the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

Close QualityUnit/web-issues#1034

### CODE REVIEW

**Comment by jremen** [comment]
@rlaco-qualityunit 
Fix the conflicts

**Comment by rlaco-qualityunit** [comment]
> @rlaco-qualityunit Fix the conflicts

@jremen fixed

File: assets/scripts/custom/compactHeader.js [hunk 1]
```diff
@@ -66,12 +70,31 @@
 		}
 		return false;
 	}
-	function fnStickyHeader() {
+	function fnStickyPlaceholder() {
 		if ( false === stickyDisable ) {
-			if ( elSticky && fnHeaderHeight() ) {
-				if ( fnHeaderHeight() <= document.documentElement.scrollTop ) {
+			if ( elSticky && elStickyPlaceholder ) {
```
Comment (jremen) [inline assets/scripts/custom/compactHeader.js:75]: merge with condition above. Avoid nesting of conditionals everywhere

File: assets/scripts/custom/compactHeader.js [hunk 2]
```diff
@@ -66,12 +70,31 @@
...
+				if ( ! elSticky.classList.contains( clStickyActive ) ) {
+					elStickyPlaceholder.style.minHeight = elSticky.offsetHeight + 'px';
+				}
+			}
+		}
+	}
+	function fnStickyHeader( scrollDirection = null ) {
+		if ( false === stickyDisable ) {
```
Comment (jremen) [inline assets/scripts/custom/compactHeader.js:83]: ! stickyDisable

File: assets/scripts/custom/compactHeader.js [hunk 3]
```diff
@@ -66,12 +70,31 @@
...
 					isSticky = true;
+				} else if ( scrollDirection === 'up' ) {
+					elSticky.classList.add( clStickyTransition );
+					setTimeout( () => {
+						elSticky.classList.remove( clStickyActive );
+						elSticky.classList.remove( clStickyTransition );
+						isSticky = false;
+						fnStickyPlaceholder();
+					}, 300 );
 				} else {
```
Comment (jremen) [inline assets/scripts/custom/compactHeader.js:98]: do not wrap into else, use straight (and end up previous if with return statement to end it)

File: assets/styles/components/_CategoryItem.scss [hunk 4]
```diff
@@ -241,7 +241,11 @@
 }
 
 .Category__item--blogLike {
+
+	@include color(border-color, border-color-gray);
 	border-radius: $border-radius-16;
+	border-style: solid;
```
Comment (jremen) [inline assets/styles/components/_CategoryItem.scss:247]: merge into border property, 1px solid xxx

File: lib/custom-blocks/compact-header.php [hunk 5]
```diff
@@ -220,13 +221,29 @@ class="compact-header__logo-img"
...
 								</svg>
 							</a>
-							<div class=" compact-header__filters-wrap
+							<?php
+							$filter_fields_count = 0;
+							if ( isset( $filer_search ) ) {
+								$filter_fields_count++;
+							}
+							if ( isset( $filer_sort ) ) {
+								$filter_fields_count++;
```
Comment (jremen) [inline lib/custom-blocks/compact-header.php:230]: use || to merge with previous conditional like if ( isset ( $filer_seach ) || isset( $filter_sort ) )

File: assets/scripts/custom/compactHeader.js [hunk 6]
```diff
@@ -66,12 +70,31 @@
 		}
 		return false;
 	}
-	function fnStickyHeader() {
+	function fnStickyPlaceholder() {
 		if ( false === stickyDisable ) {
```
[the remark this answers is not in the document]
Comment (rlaco-qualityunit) [reply assets/scripts/custom/compactHeader.js]: done

File: assets/scripts/custom/compactHeader.js (hunk 1 above)
Comment (rlaco-qualityunit) [reply assets/scripts/custom/compactHeader.js]: done

File: assets/scripts/custom/compactHeader.js (hunk 2 above)
Comment (rlaco-qualityunit) [reply assets/scripts/custom/compactHeader.js]: done

File: assets/scripts/custom/compactHeader.js [hunk 7]
```diff
@@ -66,12 +70,31 @@
...
+	}
+	function fnStickyHeader( scrollDirection = null ) {
+		if ( false === stickyDisable ) {
+			if ( elSticky && elStickyPlaceholder && fnHeaderHeight() ) {
+				fnStickyPlaceholder();
+				const stickyPoint = elStickyPlaceholder.offsetHeight * 2;
+				if ( stickyPoint <= document.documentElement.scrollTop ) {
 					elSticky.classList.add( clStickyActive );
 					isSticky = true;
+				} else if ( scrollDirection === 'up' ) {
```
[the remark this answers is not in the document]
Comment (rlaco-qualityunit) [reply assets/scripts/custom/compactHeader.js]: done

File: assets/scripts/custom/compactHeader.js (hunk 3 above)
Comment (rlaco-qualityunit) [reply assets/scripts/custom/compactHeader.js]: done

File: assets/styles/components/_CategoryItem.scss (hunk 4 above)
Comment (rlaco-qualityunit) [reply assets/styles/components/_CategoryItem.scss]: done

File: lib/custom-blocks/compact-header.php (hunk 5 above)
Comment (rlaco-qualityunit) [reply lib/custom-blocks/compact-header.php:225]: done

**Review by jremen — APPROVED** [review]

**Pull request merged**